unlock-app/src/components/interface/locks/Manage/elements/Members.tsx

```diff
@@ -0,0 +1,123 @@
+import { useQueries } from 'react-query'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { useStorageService } from '~/utils/withStorageService'
+import { useWalletService } from '~/utils/withWalletService'
+import { ToastHelper } from '~/components/helpers/toast.helper'
+import { useWeb3Service } from '~/utils/withWeb3Service'
+import { ImageBar } from './ImageBar'
+import { MemberCard } from './MemberCard'
+import useEns from '~/hooks/useEns'
+import { addressMinify } from '~/utils/strings'
+
+interface MembersProps {
+  lockAddress: string
+  network: number
+  filters?: {
+    [key: string]: any
+  }
+}
+
+export const Address = ({ address }: { address: string }) => {
+  const addressToEns = useEns(address)
+
+  const resolvedAddress =
+    addressToEns === address ? addressMinify(address) : addressToEns
+
+  return <>{resolvedAddress}</>
+}
+
+export const Members = ({
+  lockAddress,
+  network,
+  filters = {
+    query: '',
+    filterKey: 'owner',
+    expiration: 'all',
+  },
+}: MembersProps) => {
+  const { account } = useAuth()
+  const walletService = useWalletService()
+  const web3Service = useWeb3Service()
+  const storageService = useStorageService()
+
+  const getLockManagerStatus = async () => {
+    return await web3Service.isLockManager(lockAddress, account!, network)
+  }
+
+  const getMembers = async () => {
+    await storageService.loginPrompt({
+      walletService,
+      address: account!,
+      chainId: network,
+    })
+    return storageService.getKeys({
+      lockAddress,
+      network,
+      filters,
+    })
+  }
+
+  const getLockVersion = async (): Promise<number> => {
+    if (!network) return 0
+    return web3Service.publicLockVersion(lockAddress, network)
+  }
+
+  const [
+    { isLoading, data: members = [] },
+    { isLoading: isLoadingVersion, data: lockVersion = 0 },
+    { isLoading: isLoadingLockManager, data: isLockManager },
+  ] = useQueries([
+    {
+      queryFn: getMembers,
+      queryKey: 'getMembers',
```

Comment:
You need something unique in the query key to catch it. getMembers is not unique. You should pass the lock address here as well inside getMembers. Ditto for the rest.